src/pages/ScholarshipForm.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+import { Container, Button } from '@material-ui/core';
+import TextFieldComp from '../components/TextFieldComp';
+
+function ScholarshipForm() {
+  const [formFieldStates, setFormFieldStates] = useState({
+    name: '',
+    deadline: new Date(),
+    description: '',
+    amountType: '',
+    amount: 0,
+    minAmount: 0,
+    maxAmount: 0,
+    website: '',
+    scholarshipType: '',
+    active: false,
+  });
+
+  function handleFieldChange(name, value) {
+    setFormFieldStates((prev) => ({ ...prev, [name]: value }));
+  }
```

Comment:
Instead of having a `TextFieldComp` component that accepts this, this could return a function that plain `TextField` accepts. E.g.

```suggestion
  function onChangeFn(name) {
    return (e) => setFormFieldStates({ ...state, name: e.target.value});
  }
```

I gave a similar example in the comment below